daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DFDLXFunctions.scala

```diff
@@ -98,6 +98,53 @@ case class DFDLXRightShift(recipes: List[CompiledDPath], argType: NodeInfo.Kind)
   }
 }
 
+case class DFDLXXor(recipes: List[CompiledDPath], argType1: NodeInfo.Kind,argType2:NodeInfo.Kind) extends FNTwoArgs(recipes) {
+  override def computeValue(arg1: DataValuePrimitive, arg2: DataValuePrimitive, dstate: DState): DataValuePrimitive = {
+
+    argType1 match {
```

Comment:
With the changes mentioned in previous comments you no longer need to pass in two types since they will always be the same. And so I think this logic all just becomes

```scala
argType match {
  case NodeInfo.Long => arg1.getLong ^ arg2.getLong
  case NodeInfo.Int => arg1.getInt ^ arg2.getInt
  case NodeInfo.Short => (arg1.getShort ^ arg2.getShort).toShort
  ...
  case NodeInfo.UnsignedLong => arg1.getBigInt.xor(arg2.getBigInt)
  case NodeInfo.UnsignedInt => arg1.getLong ^ arg2.getLong
}
```
You don't have to do any nesting of match/cases because DPath will implicitly cast everything to the correct type.

Some things worth pointing out:
1. I don't think you need to do anything fancy for unsigned types like we had to do for the shifting functions. This is because the high order bits are all zeros for unsinged types and so we will always stay zero for the xor. So I think this can just use normal xor'slike you have
1. For types that are reprsented by java primitives smaller than an Int (i.e. short, byte, unsignedByte), we need to make sure to cast the result to that type (e.g. short needs a toShort call). This is because in Java, the result of an xor always raises the type to an integer, but the result of the function needs to be the same as the original type.